analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -304,25 +304,47 @@ fun PackageManager.resolveNuGetDependencies(
 ): ProjectAnalyzerResult {
     val workingDir = definitionFile.parentFile
 
-    val dependencies = sortedSetOf<PackageReference>()
-    val scope = Scope("dependencies", dependencies)
-
     val packages = sortedSetOf<Package>()
     val issues = mutableListOf<OrtIssue>()
 
     val references = reader.getDependencies(definitionFile)
-    val ids = references.map { Identifier(type = "NuGet", namespace = "", name = it.name, version = it.version) }
-    support.buildDependencyTree(ids, dependencies, packages, issues)
+    val referencesByFramework = references.groupBy { it.targetFramework }
+
+    val scopes = referencesByFramework.flatMapTo(sortedSetOf()) { (targetFramework, frameworkDependencies) ->
+        frameworkDependencies.groupBy { it.developmentDependency }.map { (isDevDependency, dependencies) ->
+            val allDependencies = buildSet {
+                addAll(dependencies)
+                // Add dependencies without a specified target framework to all scopes.
+                addAll(referencesByFramework[""].orEmpty().filter { it.developmentDependency == isDevDependency })
```

Comment:
For readability, `referencesByFramework[""].orEmpty()` could be extracted to a variable before the loops, e.g. to line 312 like `val dependenciesForAllFrameworks = referencesByFramework[""].orEmpty()`.